lib/spack/spack/version.py

```diff
@@ -456,6 +492,45 @@ def intersection(self, other):
 
 class GitVersion(VersionBase):
     """Class to represent versions interpreted from git refs.
+    
+    If the ref is a commit hash then there are two paths of execution:
+    
+    1) a raw comit hash needs a commit lookup to pair it to a spack version
+    2) a commit hash paired with a valid spack version  i.e. [hash]=main
+
```

Comment:
```suggestion
    There are two distinct categories of git versions:
    
    1) GitVersions instantiated with an associated reference version (e.g. 'git.foo=1.2')
    2) GitVersions requiring commit lookups

```

Reply:
I'm not positive `git.foo=1.0` works. I should test this in #30998